### PR TITLE
Cleanup warnings

### PR DESCRIPTION
**Pull request by Skallwar**
To merge after #141
- misc: Cleanup warnings
- misc: Cleanup crates

### CODE REVIEW

File: hal_aarch64/src/lib.rs [hunk 1]
```diff
@@ -13,6 +12,7 @@ pub mod mm;
 
 mod devices;
 
+#[allow(dead_code)]
```
Comment (CohenArthur) [inline hal_aarch64/src/lib.rs:15]: uff, why is that necessary?

**Review by CohenArthur — APPROVED** [review]

File: kernel/src/utils/mod.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-pub mod init_cell;
-pub mod init_once;
+// pub mod init_cell;
+// pub mod init_once;
```
[the remark this answers is not in the document]
Comment (Skallwar) [reply kernel/src/utils/mod.rs:2]: Yes, thanks

File: hal_aarch64/src/lib.rs (hunk 1 above)
Comment (Skallwar) [reply hal_aarch64/src/lib.rs:15]: We are using debug to display this later on. But the fields are not considered as accessed so we have an unused warning which is not true.

We could put a '_' but that would make the display ugly

**Review by n1tram1 — APPROVED** [review]
